chunky/src/java/se/llbit/math/PackedOctree.java

```diff
@@ -393,17 +382,11 @@ private void storeNode(DataOutputStream out, int nodeIndex) throws IOException {
     if(type == BRANCH_NODE) {
       out.writeInt(type);
       for(int i = 0; i < 8; ++i) {
-        int childIndex = treeData[nodeIndex] + 2 * i;
+        int childIndex = treeData[nodeIndex] + i;
         storeNode(out, childIndex);
       }
     } else {
-      boolean isDataNode = (treeData[nodeIndex + 1] != 0);
-      if(isDataNode) {
-        out.writeInt(type | DATA_FLAG);
-        out.writeInt(treeData[nodeIndex + 1]);
-      } else {
-        out.writeInt(type);
-      }
+      out.writeInt(type);
```

Comment:
Minor style nitpicking, the `out.writeInt(type)` could be done out side of the `if`

Reply:
That would write it in both cases because we don't return in `if (isDataNode)`

Reply:
Yeah, the same line is in both branches (and the if branch does additional stuff)